enterprise/utils.py

```diff
@@ -1115,9 +1115,13 @@ def parse_datetime_handle_invalid(datetime_value):
     """
     Return the parsed version of a datetime string. If the string is invalid, return None.
     """
+    if not datetime_value:
+        return None
     try:
         if not isinstance(datetime_value, datetime.datetime):
             datetime_value = parse_datetime(datetime_value)
+        if not datetime_value:
```

Comment:
parse_datetime can also return None when input is not formatted, hence we need this guard too

Reply:
Seems like it would be more Pythonic to use try/except blocks here versus if blocks, but I think the same end result is achieved so that's more of a style nit than anything